### PR TITLE
fix: batch create account logger and perf optimize

### PR DESCRIPTION
**Pull request by sidmorizon**

### CODE REVIEW

**Review by originalix — DISMISSED** [review]

**Review by originalix — DISMISSED** [review]

File: packages/core/src/chains/btc/CoreChainSoftware.ts [hunk 1]
```diff
@@ -781,20 +804,26 @@ export default class CoreChainSoftwareBtc extends CoreChainApiBase {
...
             addressEncoding,
           });
+        defaultLogger.account.accountCreatePerf.xpubToAddressDone();
+
         const { [firstAddressRelPath]: address } = addressesMap;
 
+        defaultLogger.account.accountCreatePerf.xpubToSegwit();
         // rebuild xpubSegwit by hd account descriptor
-        xpubSegwit = buildBtcXpubSegwit({
+        xpubSegwit = await buildBtcXpubSegwit({
```
Comment (sidmorizon) [inline packages/core/src/chains/btc/CoreChainSoftware.ts]: rename buildBtcXpubSegwitAsync

File: packages/core/src/chains/btc/sdkBtc/index.ts [hunk 2]
```diff
@@ -518,13 +518,13 @@ export function buildBtcXpubSegwit({
     // https://github.com/trezor/blockbook/blob/master/docs/api.md#get-xpub
     if (hdAccountPayload) {
       const { curveName, hdCredential, password, path } = hdAccountPayload;
-      const rootFingerprint = generateRootFingerprint(
+      const rootFingerprint = await generateRootFingerprintHex({
```
Comment (sidmorizon) [inline packages/core/src/chains/btc/sdkBtc/index.ts]: generateRootFingerprintHexAsync

Comment (sidmorizon) [inline packages/core/src/chains/btc/sdkBtc/index.ts]: generateRootFingerprintHexAsync

File: packages/core/src/chains/btc/sdkBtc/index.ts [hunk 3]
```diff
@@ -634,7 +634,7 @@ export type IGetAddressFromXpubResult = {
   publicKeys: Record<string, string>;
   xpubSegwit: string;
 };
-export function getAddressFromXpub({
+export async function getAddressFromXpub({
```
Comment (sidmorizon) [inline packages/core/src/chains/btc/sdkBtc/index.ts:637]: getAddressFromXpubAsync

File: packages/core/src/chains/btc/sdkBtc/index.ts [hunk 4]
```diff
@@ -660,7 +660,7 @@ export function getAddressFromXpub({
     encoding = supportEncodings[0];
   }
 
-  const xpubSegwit = buildBtcXpubSegwit({
+  const xpubSegwit = await buildBtcXpubSegwit({
```
Comment (sidmorizon) [inline packages/core/src/chains/btc/sdkBtc/index.ts]: buildBtcXpubSegwitAsync

File: packages/core/src/secret/encryptors/aes256.ts [hunk 5]
```diff
@@ -31,28 +34,51 @@ function keyFromPasswordAndSalt(password: string, salt: Buffer): Buffer {
 
 export const encodeKeyPrefix =
   'ENCODE_KEY::755174C1-6480-401A-8C3D-84ADB2E0C376::';
-const encodeKey = `${encodeKeyPrefix}${generateUUID()}`;
-const ENCODE_TEXT_PREFIX =
-  'SENSITIVE_ENCODE::AE7EADC1-CDA0-45FA-A340-E93BEDDEA21E::';
+let encodeKey = platformEnv.isWebEmbed
+  ? ''
```
Comment (sidmorizon) [inline packages/core/src/secret/encryptors/aes256.ts:38]: encodedKey on webemebed will be update later by native

File: packages/core/src/secret/encryptors/aes256.ts [hunk 6]
```diff
@@ -31,28 +34,51 @@ function keyFromPasswordAndSalt(password: string, salt: Buffer): Buffer {
...
+}
 
 function decodePassword({
   password,
   key,
+  ignoreLogger,
 }: {
   password: string;
   key?: string;
+  ignoreLogger?: boolean;
```
Comment (sidmorizon) [inline packages/core/src/secret/encryptors/aes256.ts:69]: 避免递归日志造成输出时序混乱

File: packages/core/src/secret/encryptors/aes256.ts [hunk 7]
```diff
@@ -229,46 +288,89 @@ function ensureSensitiveTextEncoded(text: string) {
...
 }
 
 function encodeSensitiveText({ text, key }: { text: string; key?: string }) {
   checkKeyPassedOnExtUi(key);
   const theKey = key || encodeKey;
+  ensureEncodeKeyExists(theKey);
   // text is already encoded
   if (isEncodedSensitiveText(text)) {
-    if (!platformEnv.isExtensionUi) {
+    if (!platformEnv.isExtensionUi && !platformEnv.isNative) {
```
Comment (sidmorizon) [inline packages/core/src/secret/encryptors/aes256.ts]: isDev

File: packages/core/src/secret/encryptors/aes256.ts (hunk 5 above)
Comment (sidmorizon) [reply packages/core/src/secret/encryptors/aes256.ts:38]: offscreen, extUI, webEmbed

File: packages/core/src/secret/encryptors/aes256.ts [hunk 8]
```diff
@@ -3,12 +3,15 @@ import * as crypto from 'crypto';
 import { AES_CBC, Pbkdf2HmacSha256 } from 'asmcrypto.js';
 
 import { IncorrectPassword } from '@onekeyhq/shared/src/errors';
```
Comment (sidmorizon) [inline packages/core/src/secret/encryptors/aes256.ts:5]: secret 模块禁止在 UI 层调用

File: packages/core/src/secret/encryptors/aes256.ts [hunk 9]
```diff
@@ -229,46 +288,89 @@ function ensureSensitiveTextEncoded(text: string) {
...
   return encodeKey;
 }
 
+function setBgSensitiveTextEncodeKey(key: string) {
+  if (platformEnv.isExtensionUi) {
+    throw new Error(
+      'Not allow to call setBgSensitiveTextEncodeKey() from extension ui',
+    );
+  }
+  if (!platformEnv.isWebEmbed) {
```
Comment (sidmorizon) [inline packages/core/src/secret/encryptors/aes256.ts:372]: offscreen

File: packages/core/src/secret/index.ts [hunk 10]
```diff
@@ -192,10 +201,7 @@ function encryptRevealableSeed({
     bufferUtils.bytesToHex(
       encryptString({
         password,
-        data: JSON.stringify({
-          entropyWithLangPrefixed: rs.entropyWithLangPrefixed,
-          seed: rs.seed,
-        }),
+        data: JSON.stringify(rs),
```
Comment (sidmorizon) [inline packages/core/src/secret/index.ts:204]: ensure rs serializable

File: packages/core/src/secret/index.ts [hunk 11]
```diff
@@ -451,35 +490,82 @@ function mnemonicFromEntropy(
...
+  defaultLogger.account.secretPerf.decryptHdCredential();
   const rs: IBip39RevealableSeed = decryptRevealableSeed({
     password,
     rs: hdCredential,
   });
-  return revealEntropyToMnemonic(
+  defaultLogger.account.secretPerf.decryptHdCredentialDone();
+
+  defaultLogger.account.secretPerf.revealEntropyToMnemonic();
+  const r = revealEntropyToMnemonic(
```
Comment (sidmorizon) [inline packages/core/src/secret/index.ts:501]: async？

File: packages/core/src/secret/index.ts [hunk 12]
```diff
@@ -451,35 +490,82 @@ function mnemonicFromEntropy(
...
+    mnemonicToSeedSync(params.mnemonic, params.passphrase),
+  );
+}
+
+export type IGenerateRootFingerprintHexAsyncParams = {
+  curveName: ICurveName;
+  hdCredential: IBip39RevealableSeedEncryptHex;
+  password: string;
+};
+async function generateRootFingerprintHex(
```
Comment (sidmorizon) [inline packages/core/src/secret/index.ts]: async

File: packages/core/src/secret/index.ts [hunk 13]
```diff
@@ -370,6 +381,34 @@ function batchGetPublicKeys(
...
   );
 }
+export type IBatchGetPublicKeysAsyncParams = {
+  curveName: ICurveName;
+  hdCredential: ICoreHdCredentialEncryptHex;
+  password: string;
+  prefix: string;
+  relPaths: Array<string>;
+};
+async function batchGetPublicKeysAsync(
```
Comment (sidmorizon) [inline packages/core/src/secret/index.ts:391]: batchGetPrivateKeysAsync

File: packages/core/src/secret/index.ts [hunk 14]
```diff
@@ -451,35 +490,82 @@ function mnemonicFromEntropy(
...
   const publicKey = publicFromPrivate(curveName, masterKey.key, password);
-  return hash160(publicKey).slice(0, 4);
+  return hash160(publicKey).slice(0, 4).toString('hex');
 }
 
 export {
-  CKDPriv,
-  CKDPub,
-  N,
   batchGetPrivateKeys,
```
Comment (sidmorizon) [inline packages/core/src/secret/index.ts:566]: 其他所有可能的函数都异步改造

File: packages/kit-bg/src/apis/backgroundApiPermissions.ts [hunk 15]
```diff
@@ -6,16 +6,24 @@ import type {
   IJsonRpcRequest,
 } from '@onekeyfe/cross-inpage-provider-types';
 
-const PRIVATE_WHITE_LIST_ORIGIN = [
-  'https://onekey.so',
-  'http://localhost:3008', // iOS simulator DEV localhost for web-embed
-  'http://localhost:8081', // iOS simulator DEV localhost for web-embed
-  'null', // Android DEV localhost for web-embed. url like file://
+export const WEB_EMBED_API_WHITE_LIST_ORIGIN = [
```
Comment (sidmorizon) [inline packages/kit-bg/src/apis/backgroundApiPermissions.ts:9]: 把 webembed 配置的 host 自动加进来，仅 isDev

File: packages/kit-bg/src/apis/backgroundApiPermissions.ts [hunk 16]
```diff
@@ -24,31 +32,43 @@ const PRIVATE_WHITE_LIST_ORIGIN = [
     : []),
 ].filter(Boolean);
 
-export function isPrivateAllowedOrigin(origin?: string) {
+export const PROVIDER_API_PRIVATE_WHITE_LIST_ORIGIN = [
+  'https://onekey.so',
```
Comment (sidmorizon) [inline packages/kit-bg/src/apis/backgroundApiPermissions.ts:37]: http://1key.so/

Comment (sidmorizon) [inline packages/kit-bg/src/apis/backgroundApiPermissions.ts:37]: http://1key.so/

File: packages/kit-bg/src/dbs/local/realm/RealmDBAgent.ts [hunk 17]
```diff
@@ -61,12 +61,12 @@ export class RealmDBAgent extends LocalDbAgentBase implements ILocalDBAgent {
     recordId: string,
   ) {
     checkIsDefined(storeName);
-    console.log('realmdb _getObjectRecordById ', { storeName, recordId });
+    // console.log('realmdb _getObjectRecordById ', { storeName, recordId });
```
Comment (sidmorizon) [inline packages/kit-bg/src/dbs/local/realm/RealmDBAgent.ts:64]: realm db log

File: packages/kit-bg/src/services/ServiceHardwareUI/ServiceHardwareUI.ts [hunk 18]
```diff
@@ -142,8 +149,19 @@ class ServiceHardwareUI extends ServiceBase {
...
-  async cleanHardwareUiState() {
+  async cleanHardwareUiState({
+    hardClose,
+  }: {
+    hardClose?: boolean; // hard close dialog by event bus
+  } = {}) {
     await hardwareUiStateAtom.set(undefined);
+    if (hardClose) {
+      // atom some times not work, emit event to hard close dialog
+      appEventBus.emit(
```
Comment (sidmorizon) [inline packages/kit-bg/src/services/ServiceHardwareUI/ServiceHardwareUI.ts:160]: use second timestamp atom to close dialog

File: packages/shared/src/storage/webEmbedConfig.ts [hunk 19]
```diff
@@ -0,0 +1,35 @@
+import { EAppSettingKey } from '@onekeyhq/shared/src/storage/appSetting';
+import appStorage from '@onekeyhq/shared/src/storage/appStorage';
+
+export type IWebEmbedConfig = {
+  debug: boolean | undefined;
+  url: string | undefined;
+};
+
+function getWebEmbedConfig(): IWebEmbedConfig {
```
Comment (sidmorizon) [inline packages/shared/src/storage/webEmbedConfig.ts]: IWebEmbedConfig | undefined

**Review by revan-zhang — APPROVED** [review]

**Review by originalix — APPROVED** [review]

**Pull request merged**